components/renku_data_services/notebooks/blueprints.py

```diff
@@ -730,13 +738,17 @@ def server_logs(self) -> BlueprintFactoryResponse:
         async def _server_logs(
             request: Request, user: AnonymousAPIUser | AuthenticatedAPIUser, server_name: str
         ) -> JSONResponse:
-            max_lines = int(request.query_args.get("max_lines", 250))
-            logs = self.nb_config.k8s_client.get_server_logs(
-                server_name=server_name,
-                max_log_lines=max_lines,
-                safe_username=user.id,
-            )
-            return json(ServerLogs().dump(logs))
+            args: dict[str, str | int] = request.get_args()
+            max_lines = int(args.get("max_lines", 250))
+            try:
+                logs = await self.nb_config.k8s_client.get_server_logs(
+                    server_name=server_name,
+                    safe_username=user.id,
+                    max_log_lines=max_lines,
+                )
+                return json(ServerLogs().dump(logs))
+            except MissingResourceError as err:
+                raise exceptions.NotFound(message=err.message)
```

Comment:
> If I understand things correctly, k8s_client should use the one from errors.errors rather than notebook.errors.user.

Yes that is exactly right. We should just fully eliminate the errors from `notebook.errors`